lambdas/build-stac/utils/events.py

```diff
@@ -1,33 +1,37 @@
+import re
 from datetime import datetime
-from typing import Dict, List, Literal, Optional, Union
 from pathlib import Path
-import re
+from typing import Dict, List, Literal, Optional, Union
 
-from pydantic import BaseModel, Field
 import pystac
-
+from pydantic import BaseModel, Field
 
 INTERVAL = Literal["month", "year"]
 
 
-class BaseEvent(BaseModel, frozen=True):
+class BaseEvent(BaseModel, frozen=True, arbitrary_types_allowed=True):
     collection: str
     remote_fileurl: str
 
+    product_id: Optional[str] = None
     id_regex: Optional[str] = None
     asset_name: Optional[str] = None
     asset_roles: Optional[List[str]] = None
     asset_media_type: Optional[Union[str, pystac.MediaType]] = None
-    assets: Optional[List[Dict]] = None
+    assets: Optional[Dict[str, pystac.Asset]] = (None,)
     mode: Optional[str] = None
     test_links: Optional[bool] = False
     reverse_coords: Optional[bool]
 
     def item_id(self: "BaseEvent") -> str:
         if self.id_regex:
-            id_components = re.findall(self.id_regex, self.s3_filename)
+            id_components = re.findall(
+                self.id_regex, self.s3_filename
+            )  # TODO: s3_filename
```

Comment:
whats the TODO?

Reply:
The todo is to either add it back in or change it - there's no self.s3_filename that I can see 🤷🏼‍♂️